src/app/shared/services/http.service.ts

```diff
@@ -110,47 +117,46 @@ export class HttpService {
       .pipe(catchError(this.handleError()));
   }
 
-  getReports(reportIds: string[], storage: string) {
+  getReports(reportIds: string[], storage: string): Observable<Record<string, CompareReport>> {
     return this.http
-      .get<any>(
-        'api/report/' + storage + '/?xml=true&globalTransformer=' + localStorage.getItem('transformationEnabled'),
-        { params: { storageIds: reportIds } },
-      )
+      .get<
+        Record<string, CompareReport>
+      >('api/report/' + storage + '/?xml=true&globalTransformer=' + localStorage.getItem('transformationEnabled'), { params: { storageIds: reportIds } })
```

Comment:
It is always possible that a cookie is empty, does this cause weird stuff when calling the endpoint? If it does, you could add a check.